File: frontend/src/types/team.ts
/**
 * Team Types
 *
 * TypeScript type definitions for team management.
 */

/**
 * Team from API
 */
export interface Team {
  id: string;
  organization_id: string;
  name: string;
  description: string | null;
  manager_id: string | null;
  created_at: string;
  updated_at: string;
}

/**
 * Team response with member count
 */
export interface TeamResponse {
  id: string;
  organization_id: string;
  name: string;
  description: string | null;
  manager_id: string | null;
  member_count: number;
  created_at: string;
  updated_at: string;
}

/**
 * Team member info
 */
export interface TeamMemberInfo {
  user_id: string;
  email: string;
  first_name: string;
  last_name: string;
  joined_at: string;
}

/**
 * Team with members
 */
export interface TeamWithMembers {
  team: Team;
  members: TeamMemberInfo[];
}

/**
 * Create team request
 */
export interface CreateTeamRequest {
  name: string;
  description?: string;
  manager_id?: string;
}

/**
 * Update team request
 */
export interface UpdateTeamRequest {
  name?: string;
  description?: string | null;
  manager_id?: string | null;
}

/**
 * Add member request
 */
export interface AddMemberRequest {
  user_id: string;
}

/**
 * Team member response
 */
export interface TeamMember {
  id: string;
  team_id: string;
  user_id: string;
  joined_at: string;
}

/**
 * List teams query parameters
 */
export interface ListTeamsParams {
  page?: number;
  per_page?: number;
  search?: string;
  manager_id?: string;
}

/**
 * Paginated teams response
 */
export interface PaginatedTeamsResponse {
  teams: TeamResponse[];
  total: number;
  page: number;
  per_page: number;
}

File: backend/src/models/team.rs
use chrono::{DateTime, Utc};
use diesel::prelude::*;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use crate::schema::teams;

/// Team entity from database
#[derive(Debug, Clone, Queryable, Selectable, Serialize, Deserialize)]
#[diesel(table_name = teams)]
#[diesel(check_for_backend(diesel::pg::Pg))]
pub struct Team {
    pub id: Uuid,
    pub organization_id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub manager_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

/// NewTeam for creating teams
#[derive(Debug, Insertable)]
#[diesel(table_name = teams)]
pub struct NewTeam {
    pub organization_id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub manager_id: Option<Uuid>,
}

/// Team update struct for partial updates
#[derive(Debug, AsChangeset, Default)]
#[diesel(table_name = teams)]
pub struct TeamUpdate {
    pub name: Option<String>,
    pub description: Option<Option<String>>,
    pub manager_id: Option<Option<Uuid>>,
    pub updated_at: Option<DateTime<Utc>>,
}

/// Team response with member count
#[derive(Debug, Serialize)]
pub struct TeamResponse {
    pub id: Uuid,
    pub organization_id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub manager_id: Option<Uuid>,
    pub member_count: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

/// Team with members for detailed view
#[derive(Debug, Serialize)]
pub struct TeamWithMembers {
    pub team: Team,
    pub members: Vec<TeamMemberInfo>,
}

/// Team member info for response
#[derive(Debug, Serialize)]
pub struct TeamMemberInfo {
    pub user_id: Uuid,
    pub email: String,
    pub first_name: String,
    pub last_name: String,
    pub joined_at: DateTime<Utc>,
}

/// Team filter options
#[derive(Debug, Clone, Default)]
pub struct TeamFilter {
    pub search: Option<String>,
    pub manager_id: Option<Uuid>,
}
